main.tf

```diff
@@ -14,7 +14,259 @@
  * limitations under the License.
  */
 
-resource "google_storage_bucket" "main" {
+locals {
+  all_vpcs = ["mgmt", "inet", "lan"]
+  regions  = tolist([for region in var.network_regions : region.name])
+
+  subnets = {
+    for vpc in local.all_vpcs : vpc =>
+    flatten([
+      for network_region in var.network_regions : {
+        subnet_name   = "${vpc}-vpc-subnet-${network_region.name}"
+        subnet_ip     = network_region["${vpc}_subnet"]
+        subnet_region = network_region.name
+      }
+    ])
+  }
+
+  network_interfaces = {
+    for key, vpc in module.sdwan_vpc :
+    key => {
+      for region in var.network_regions : region.name => {
+        network       = vpc.network.network_self_link
+        subnetwork    = vpc.subnets["${region.name}/${vpc.network_name}-subnet-${region.name}"].self_link
+        access_config = (key == "inet" ? [{}] : [])
+        network_ip    = (key == "lan" ? google_compute_address.lan_subnet_static_ip["${region.name}"].address : "")
+      }
+    }
+  }
+}
+
+module "sdwan_vpc" {
+  for_each = toset(local.all_vpcs)
+  source   = "terraform-google-modules/network/google"
+  version  = "~> 5.0"
+
+  project_id   = var.project_id
+  network_name = "${each.value}-vpc"
+  routing_mode = "GLOBAL"
+
+  subnets = local.subnets[each.value]
+}
+
+module "inet_firewall_rules" {
+  source       = "terraform-google-modules/network/google//modules/firewall-rules"
+  project_id   = var.project_id
+  network_name = module.sdwan_vpc["inet"].network_name
+
+  rules = [{
+    name                    = "inet-vpc-allow-vcmp-ingress"
+    description             = null
+    direction               = "INGRESS"
+    priority                = null
+    ranges                  = ["0.0.0.0/0"]
+    source_tags             = null
+    source_service_accounts = null
+    target_tags             = null
+    target_service_accounts = null
+    allow = [{
+      protocol = "udp"
+      ports    = ["2426"]
+    }]
+    deny = []
+    log_config = {
+      metadata = "EXCLUDE_ALL_METADATA"
+    }
+  }]
+}
+
+module "lan_firewall_rules" {
+  source       = "terraform-google-modules/network/google//modules/firewall-rules"
```

Comment:
perhaps makes sense to pin the version?

Reply:
yup, will add a version constraint here.